### PR TITLE
Bugfix when reading the pins and the PIC chip is still writing.

### PR DESCRIPTION
**Pull request by Mark--S**
Sometimes when we try to make a PIN reading the PIC chip hasn't finished writing to the buffer, so we only get part of the PIN reading. This bugfix makes the server wait for 0.1 seconds if it doesn't get a complete PIN reading. After the wait the code reads out the buffer again appending it to whatever we already had.

### CODE REVIEW

**Comment by Mark--S** [comment]
This needs testing underground before merge.

**Comment by mnirkko** [comment]
Can confirm code seems to work, no more warnings occurred over approx. 2M events. (Previously was about 1 in 10^5 events.)

File: core/tellie_server.py [hunk 1]
```diff
@@ -516,8 +516,24 @@ def read_pin_sequence(self):
         if len(numbers) == 0:
             self.log_phrase("Sequence doesn't appear to have finished..", 0, _snotDaqLog)
             return None
-        elif len(numbers) == 2:
+        #If we only get one number it is likely the PIC hasnt finished writing to the buffer
+        if len(numbers) == 1:
+            #Wait a bit longer and then read out the buffer again
```
Comment (EdLeming) [inline core/tellie_server.py]: I think this wants to be in a try catch incase there is a case where the PIC outputs a single number, but isn't in the middle of a sequence.

**Comment by Mark--S** [comment]
Okay I think Ive made all the changes.

**Comment by EdLeming** [comment]
Cool, looks good to me. Lemme know when it's been tested and I'll merge

**Comment by mnirkko** [comment]
So the only change seems to be an extra delay, and slightly different output. I can probably run TELLIE locally, but not via ORCA as the timing rack is currently off (we are working on the laser trigger system today). Would you be ok with that?

**Comment by EdLeming** [comment]
yeah, that's fine 👍

**Comment by mnirkko** [comment]
Seems to work ok. Can run in Master mode at maximal pulse number for different channels (upper and lower boxes).

**Comment by mnirkko** [comment]
So the other PR #46 appears to be a red herring. The fixes in this PR are fine, and I haven't got any of the above warnings from running locally.

**Pull request merged**